### PR TITLE
Update conditional requirement for PySide

### LINKED ISSUE #2577: PySide not compatible with lastest python versions
looks like that pyside, in the opt deps, is compatible only with a limited range of python versions.

```
pip install PySide --no-cache-dir
Collecting PySide
  Downloading PySide-1.2.4.tar.gz (9.3MB)
    100% |████████████████████████████████| 9.3MB 1.5MB/s
    Complete output from command python setup.py egg_info:
    only these python versions are supported: [(2, 6), (2, 7), (3, 2), (3, 3), (3, 4)]

```

Therefore, pip install -r optional_requirements.txt can fail.

### PR DESCRIPTION
**Pull request by sciunto**
## Description
Closes #2577

### CODE REVIEW

File: optional_requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
-PySide; python_version != '2.7'
+PySide; python_version != '3.5' and python_version != '3.6'
```
Comment (soupault) [inline optional_requirements.txt]: Why remove 2.7?

Comment (sciunto) [inline optional_requirements.txt]: Because pyside is compatible with py2.7 Do I miss why it was blacklisted?

Comment (soupault) [inline optional_requirements.txt]: I'm not aware. Maybe for CI purposes...

File: optional_requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 imread; python_version != '2.7'
-SimpleITK; python_version != '3.4' and python_version != '3.5'
+SimpleITK; python_version != '3.4' and python_version != '3.5' and python_version != '3.6'
```
Comment (soupault) [inline optional_requirements.txt]: SimpleITK seems to have a support for 3.4: https://pypi.python.org/pypi/SimpleITK/0.9.1 .

**Comment by sciunto** [comment]
I'm not available over the week-end. If you want to tweak that before Monday, feel free. Edits are allowed on the PR.

**Comment by sciunto** [comment]
updated. @scikit-image/packaging Any advice?

**Pull request merged**